src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -125,9 +120,44 @@ create_distributed_table(PG_FUNCTION_ARGS)
 	text *distributionColumnText = PG_GETARG_TEXT_P(1);
 	Oid distributionMethodOid = PG_GETARG_OID(2);
 
+	text *colocateWithTableNameText = PG_GETARG_TEXT_P(3);
+	char *colocateWithTableName = text_to_cstring(colocateWithTableNameText);
+
 	char *distributionColumnName = text_to_cstring(distributionColumnText);
 	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
 
+	/* check if we try to colocate with hash distributed tables */
+	if (strncmp(colocateWithTableName, "default", NAMEDATALEN) != 0 &&
+		strncmp(colocateWithTableName, "none", NAMEDATALEN) != 0)
```

Comment:
I think that would makes everything more confusing. (i.e. "none" as special keyword and "NONE" as table name) and what about "None"?

My suggestion would be;

none, NONE, nOnE... => all these are special keywords.
schema_name.{none | NONE | nOnE} => all these will refer tables. something like;

```sql
create_distributed_table('partsupp', 'ps_partkey', colocate_with => 'current_schema_name.nONe');
```

Would that makes sense?